src/components/Loading.js

```diff
@@ -0,0 +1,42 @@
+import { dispatch } from "utils/dispatch";
+import {
+  // defaultMapConfig,
+  TOTAL_NUMBER_OF_MAP_LAYERS,
+} from "utils/constants";
+
+export class Loading {
+  //laodng screen container
+  loadingScreen = document.getElementById("loading-container");
+
+  // loaded layers counter
+  layersLoaded = 0;
+
+  // optionally pass in the mapconfig to keep
+  // track of which layers are set to true
+  constructor(/*config*/) {
+    // this.config = config || defaultMapConfig;
+    this.bindListeners();
+  }
+
+  bindListeners() {
+    //layer can be passed in here to keep track of
+    //which layers have loaded
+    dispatch.on("fetch-map-data-resolve", (/*layer*/) => {
```

Comment:
It would probably make sense to name space this event, e.g. `"fetch-map-data-resolve.loading"` as not name spacing it will prevent the event from not being registered elsewhere if another listener also uses `"fetch-map-data-resolve"` without a name space. See the readme for d3-dispatch for more.

Reply:
Definitely all valid comments especially on the re-usability.  Will work on this one.